symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/lib/jsonmapper/JsonMapperTest.java

```diff
@@ -0,0 +1,154 @@
+  }
+
+  @Test
+  public void toEnricherStringTest() {
+    String result = jsonMapper.toEnricherString("User Details", userDetails, "1.0.0");
+    assertEquals(result, "{"
+        + "\"User Details\":{"
+        + "\"type\":\"User Details\",\"version\":\"1.0.0\",\"payload\":\"{"
+        + "\\\"userId\\\":123456,\\\"email\\\":\\\"testuser@email.com\\\",\\\"firstName\\\":\\\"John\\\","
+        + "\\\"lastName\\\":\\\"Doe\\\",\\\"displayName\\\":\\\"Test User\\\",\\\"username\\\":\\\"Test User\\\""
+        + "}\"}}");
+  }
+
+  @Test
+  public void objectToMapTest() {
+    Map<String, Object> result = jsonMapper.objectToMap(userDetails);
+    assertEquals(result.get("userId"), 123456L);
+    assertEquals(result.get("username"), "Test User");
+    assertEquals(result.get("firstName"), "John");
+    assertEquals(result.get("lastName"), "Doe");
+    assertEquals(result.get("displayName"), "Test User");
+    assertEquals(result.get("email"), "testuser@email.com");
+  }
+
+  @Test
+  public void toObjectTest() {
+    UserDetails result = jsonMapper.toObject(
+        userDetailJson,
+        UserDetails.class
+    );
+    assertEquals(result, userDetails);
+  }
+
+  @Test
+  public void toObjectListTest() {
+    String usersListJson = "["
+        + "{\"userId\":123456,\"email\":\"testuser@email.com\",\"firstName\":\"John\",\"lastName\":\"Doe\",\"displayName\":\"Test User\",\"username\":\"Test User\"},"
+        + "{\"userId\":456123,\"email\":\"botuser@email.com\",\"firstName\":\"Bot\",\"lastName\":\"User\",\"displayName\":\"Bot User\",\"username\":\"Bot User\"},"
+        + "{\"userId\":654321,\"email\":\"testuser2@email.com\",\"firstName\":\"Test\",\"lastName\":\"User\",\"displayName\":\"Test 2\",\"username\":\"Test 2\"}"
+        + "]";
+    UserDetails[] result = jsonMapper.toObject(
+        usersListJson,
+        UserDetails[].class
+    );
+    assertEquals(result[0].getUserId(), 123456L);
+    assertEquals(result[0].getUsername(), "Test User");
+    assertEquals(result[0].getFirstName(), "John");
+    assertEquals(result[0].getLastName(), "Doe");
+    assertEquals(result[0].getDisplayName(), "Test User");
+    assertEquals(result[0].getEmail(), "testuser@email.com");
+
+    assertEquals(result[1].getUserId(), 456123L);
+    assertEquals(result[1].getUsername(), "Bot User");
+    assertEquals(result[1].getFirstName(), "Bot");
+    assertEquals(result[1].getLastName(), "User");
+    assertEquals(result[1].getDisplayName(), "Bot User");
+    assertEquals(result[1].getEmail(), "botuser@email.com");
+
+    assertEquals(result[2].getUserId(), 654321L);
+    assertEquals(result[2].getUsername(), "Test 2");
+    assertEquals(result[2].getFirstName(), "Test");
+    assertEquals(result[2].getLastName(), "User");
+    assertEquals(result[2].getDisplayName(), "Test 2");
+    assertEquals(result[2].getEmail(), "testuser2@email.com");
+  }
+
+  @Test
+  public void toObjectFailedTest() throws JsonProcessingException {
+    when(objectMapper.readValue(
```

Comment:
You could as well add a test with an bad formatted json string and check it throws the exception.